docs/_source/guides/llms/practical_guides/fine_tune.md

```diff
@@ -1,24 +1,305 @@
+|:-----------------------------------|:-----------------|:-----------------------------------------------------------|:------------------|
+| for_text_classification            | `text-label`     | `Union[Tuple[str, str], Tuple[str, List[str]]]`            | ✔️                 |
+| for_supervised_fine_tuning         | `text`           | `str`                                                      | ✗                 |
+| for_reward_modelling               | `chosen-rejected`| `Tuple[str, str]`                                          | ✗                 |
+| for_proximal_policy_optimization   | `text`           | `Union[str, Iterator[str]]`                                | ✗                 |
+| for_direct_performance_optimization| `prompt-chosen-rejected`                 | `Tuple[str, str]`                                          | ✗                 |
+
+
+## Tasks
+
+### Text Classification
+
+#### Background
+
+Text classification is a widely used NLP task where labels are assigned to text. Major companies rely on it for various applications. Sentiment analysis, a popular form of text classification, assigns labels like 🙂 positive, 🙁 negative, or 😐 neutral to text. Additionally, we distinguish between single- and multi-label text classification.
+
+::::{tab-set}
+
+:::{tab-item} Single-label
+Single-label text classification refers to the task of assigning a single category or label to a given text sample. Each text is associated with only one predefined class or category. For example, in sentiment analysis, a single-label text classification task would involve assigning labels such as "positive," "negative," or "neutral" to texts based on their sentiment.
+
+```batch
+"The help for my application of a new card and mortgage was great", "positive"
+```
+
+:::
+
+:::{tab-item} Multi-label
+Multi-label text classification is generally more complex than single-label classification due to the challenge of determining and predicting multiple relevant labels for each text. It finds applications in various domains, including document tagging, topic labeling, and content recommendation systems. For example, in customer care, a multi-label text classification task would involve assigning topics such as "new_card," "mortgage," or "opening_hours" to texts based on their content.
+
+```{tip}
+For a multi-label scenario it is recommended to add some examples without any labels to improve model performance.
+```
+
+```batch
+"The help for my application of a new card and mortgage was great", ["new_card", "mortgage"]
+```
+
+:::
+
+::::
+
+#### Training
 
-### Data
+Text classification is one of the most widely supported training tasks tasks within NLP. For example purposes we will use our [Stackoverflow demo dataset](https://huggingface.co/datasets/argilla/stackoverflow_feedback_demo).
+
+**Data Preparation**
+
+```python
+from argilla.feedback import FeedbackDataset
+
+dataset = FeedbackDataset.from_huggingface(
+    repo_id="argilla/emotion"
```

Comment:
```suggestion
    repo_id="argilla/stackoverflow_feedback_demo"
```
Then we should use the Stack overflow demo right? Or is the text above wrong?

Reply:
Or the other way around, that we say we'll use the emotion dataset for the demo.